src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -479,75 +527,33 @@ public void traverse() {
-            } catch (final IOException e) {
-                throw new UserException("Error initializing GenomicsDBImporter in batch " + batchCount, e);
-            } catch (final IllegalArgumentException iae) {
-                throw new GATKException("Null feature reader found in sampleNameMap file: " + sampleNameMapFile, iae);
-            }
-            try {
-                importer.importBatch();
-            } catch (final IOException e) {
-                throw new UserException("GenomicsDB import failed in batch " + batchCount, e);
-            }
-            closeReaders(sampleToReaderMap);
-            progressMeter.update(intervals.get(0));
-            logger.info("Done importing batch " + batchCount + "/" + totalBatchCount);
+        try {
+            importer = new GenomicsDBImporter(importConfig);
+            importer.executeImport();
+        } catch (final IOException e) {
+            throw new UserException("Error initializing GenomicsDBImporter", e);
+        } catch (final IllegalArgumentException iae) {
+            throw new GATKException("Null feature reader found in sampleNameMap file: " + sampleNameMapFile, iae);
+        } catch (final InterruptedException ie) {
+            throw new GATKException("Error while executing a parallel import in GenomicsDBImporter", ie);
         }
     }
 
     @Override
     public Object onTraversalSuccess() {
-        if (batchSize==DEFAULT_ZERO_BATCH_SIZE) {
+        if (batchSize == DEFAULT_ZERO_BATCH_SIZE) {
             logger.info("Import completed!");
         } else {
             logger.info("Import of all batches to GenomicsDB completed!");
         }
-
-        // Write the vid and callset map JSON files
-        try {
-            GenomicsDBImporter.writeVidMapJSONFile(vidMapJSONFile.getAbsolutePath(), mergedHeaderLines);
-        } catch (final FileNotFoundException fe) {
-            throw new UserException("Unable to write vid map JSON file " + vidMapJSONFile.getAbsolutePath(), fe);
-        }
-        try {
-            GenomicsDBImporter.writeCallsetMapJSONFile(callsetMapJSONFile.getAbsolutePath(), callsetMappingPB);
-        } catch (final FileNotFoundException fe) {
-            throw new UserException("Unable to write callset map JSON file " + callsetMapJSONFile.getAbsolutePath(), fe);
-        }
-        try {
-            GenomicsDBImporter.writeVcfHeaderFile(vcfHeaderFile.getAbsolutePath(), mergedHeaderLines);
-        } catch (final FileNotFoundException fe) {
-            throw new UserException("Unable to write VCF Header file " + vcfHeaderFile.getAbsolutePath(), fe);
-        }
-
-
-        if (doConsolidation) {
-            logger.info("GenomicsDB consolidation started");
-            GenomicsDBImporter.consolidateTileDBArray(workspace, GenomicsDBConstants.DEFAULT_ARRAY_NAME);
-            logger.info("GenomicsDB consolidation completed");
-        }
```

Comment:
I'm assuming that consolidation and writing the JSONs are handled internally now?

Reply:
Yes